### ISSUE TITLE
Script installation se4-fs à finaliser 

### ISSUE THREAD

**Issue by mollef** [SambaEdu/se4#8]
https://github.com/SambaEdu/sambaedu-config/blob/master/sources/var/www/diconf/install_se4fs_phase2.sh

* Interaction avec les paquets sambaEdu et passage des pramètres

**Comment by denis-b** [comment]
# Les prérequis minimaux pour une installation totalement automatique de sambaedu en stretch sont : 

- sources apt non signées acceptées pour se4XP
- connectivité réseau complète, avec se4ad configuré comme serveur dns, et ntp
- /etc/sambaedu/sambaedu.conf minimal
- ssh root@se4ad sans mot de passe

En cas de migration d'un se3 : 

- reservations dhcp dans /etc/sambaedu/
- /etc/sambaedu/sambaedu.conf.d/dhcp.conf 
- dump des tables se3db.connexions, se3db.quotas, dans /etc/sambaedu

Dans ces conditions, 
apt install sambaedu -y devrait configurer le système sans trop poser de questions.

# comment passer les fichiers 
- preseed : http au boot, ou lors de l'installation d-i
- install_se4fs_phase2.sh et .profile : récupérée lors de la phase 1
- autres fichiers : archive .tgz, récupérée lors de la phase 1

comment faire si le réseau configuré n'accède pas au se3 ?

**Comment by mollef** [comment]
ok actuellement l'installeur généère un fichier du nom de se4fs.conf, je vais donc renommer le fichier en question.

Par ailleurs, il contient les infos suivantes,

 les variables sont elles concordantes avec celles des paquets et faut-il en ajouter d'autres ?

A noter que "ntpserv" sera modifier afin de pointer sur l'ip de SE4AD - A voir quoi modifier au niveau systemD pour prise en compte ou bien un des paquets s'en charge ?
```
cat /var/www/diconf/se4fs.config 
## Adresse IP du futur SE4-AD ##
se4ad_ip="10.127.164.205"
## Adresse IP du futur SE4-FS ##
se4fs_ip="10.127.164.210"
## Nom du futur SE4-FS ##
se4fs_name="se4fs"
## Nom de domaine samba du SE4-AD ##
smb4_domain="stage"
## Suffixe du domaine##
suffix_domain="ac-rouen.fr"
## Nom de domaine complet - realm du SE4-AD ##
ad_domain="stage.ac-rouen.fr"
##Adresse du serveur DNS##
nameserver="10.127.164.1"
##SID domaine actuel
domainsid="S-1-5-21-3271951266-3673128075-3782327119"
##NTP server 
ntpserv="chronos.ac-rouen.fr"

```

**Comment by denis-b** [comment]
Le 18/05/2018 à 15:07, Franck Molle a écrit :
>
> ok actuellement l'installeur généère un fichier du nom de se4fs.conf, 
> je vais donc renommer le fichier en question.
>
> Par ailleurs, il contient les infos suivantes,
>
> les variables sont elles concordantes avec celles des paquets et 
> faut-il en ajouter d'autres ?
>
> A noter que "ntpserv" sera modifier afin de pointer sur l'ip de SE4AD 
> - A voir quoi modifier au niveau systemD pour prise en compte ou bien 
> un des paquets s'en charge ?
>
> |cat /var/www/diconf/se4fs.config ## Adresse IP du futur SE4-AD ## 
> se4ad_ip="10.127.164.205" ## Adresse IP du futur SE4-FS ## 
> se4fs_ip="10.127.164.210" ## Nom du futur SE4-FS ## se4fs_name="se4fs" 
> ## Nom de domaine samba du SE4-AD ## smb4_domain="stage" ## Suffixe du 
> domaine## suffix_domain="ac-rouen.fr" ## Nom de domaine complet - 
> realm du SE4-AD ## ad_domain="stage.ac-rouen.fr" ##Adresse du serveur 
> DNS## nameserver="10.127.164.1" ##SID domaine actuel 
> domainsid="S-1-5-21-3271951266-3673128075-3782327119" ##NTP server 
> ntpserv="chronos.ac-rouen.fr" |

|C'est presque cela :

Voici le minimum pour être silencieux:

admin_name="Administrator"
se4ad_ip="192.168.200.3"
se4ad_name="se3"
samba_domain="sambaedu3" # smb4_domain
domain="sambaedu3.maison" # ad_domain
se4fs_ip="192.168.200.2"
ldap_base_dn="dc=sambaedu3,dc=maison"
ldap_admin_name="Administrator"
ldap_admin_passwd="wwwwwww"


Pour tout le reste de la conf initiale, surtout ne rien faire, les 
paquets sont autosuffisants, se débrouillent tout seuls sur une stretch 
d'origine.


pour sambaedu-web, les paquets installent l'interface web, avec quelques 
pages fonctionnelles.  C'est la branche debconf du git qui est à jour.



Bon WE !

Denis

|

**Comment by mollef** [comment]
Ok parfait à part le mot de passe, je peux tout passer dans la conf.
Quelle différence entre admin_name et  ldap_admin_name ? Sous AD c'est la même chose non ?
Bon W.E
Franck

**Comment by denis-b** [comment]
Effectivement, c'est un bug... J'ai dû oublier de le changer quelque part, et donc il y a les 2 qui sont créés. Le bon c'est ldap_admin_name

**Comment by mollef** [comment]
Message de Denis remis dans la bonne issue.
En résumé :
* rien dans /etc/sambaedu sauf les fichiers de conf, ou alors je filtre mieux les fichiers. :+1: *.conf par exemple
* people_rdn et ses freres doit être dans le conf. SI le fichier est vide, il prend le modèle et c'est ok. Si on le passe prérenseigné, il faut qu'ils y soient. 
* priorité trop basse de certains questions, il faut qu'en cas de pb les questions soient posées. Je vais changer le niveau des questions en cas d'erreur. 

le plus important  : 
* il faut impérativement créer l'utilisateur  service www-sambaedu sur se4ad lors de la crétion de se4ad, et exporter son keytab. sinon rien ne pourra marcher... 
``` samba-tool user create www-sambaedu --description="Utilisateur admin de l'interface web" --random-password samba-tool group addmembers "Domain Admins" www-sambaedu ```
``` samba-tool domain exportkeytab --principal=www-sambaedu@SAMBAEDU3.MAISON /root/www-sambaedu.keytab  ``` 
* Echange des clés ssh avant entre les deux serveurs car le keytab est normalement copié en ssh lors du postinst sambaedu-web-common.

**Comment by denis-b** [comment]
https://github.com/SambaEdu/se4/commit/48cbed48ff2dd4625f3eb3fa95af266d36b6ac37

Le domaine est correctement récupéré. 
En revanche la présence du keytab www-sambaedu sur se4ad est indispensable.

Petit point de détail : le se4ad doit impérativement s'appeler se4ad : est-ce un problème ?

**Comment by denis-b** [comment]
En fait la résolution inverse dns fonctionne bizarrement avec samba-tool en mode kerberos : on ne peut donc pas faire de "samba-tool dns add $se4ad_ip $domain ..... " On va faire autrement...

**Comment by denis-b** [comment]
Il faut également prévoir l'export des tables sql à récuperer : quota, connexions.
Il faut voir où on les met. /root, /etc/sambaedu ?

**Comment by mollef** [comment]
dans /etc/sambaedu ?
Il y a aussi les réservations dhcp qui sont récupérées et envoyées dans /etc/sambaedu/ du se4fs.  Après il faut voir s'il n'y a pas un risque de lecture et interprétation de config.inc.sh de ces fichiers.

Faut-il récupérer la conf du dhcp dhcpd.conf car tu y fais mention... Est-ce que ça fait pas doublon avec reservation.conf ?

**Comment by denis-b** [comment]
Le 25/05/2018 à 09:03, Franck Molle a écrit :
>
> dans /etc/sambaedu ?
> Il y a aussi les réservations dhcp qui sont récupérées et envoyées 
> dans /etc/sambaedu/ du se4fs. Après il faut voir s'il n'y a pas un 
> risque de lecture et interprétation de config.inc.sh de ces fichiers.
>
> Faut-il récupérer la conf du dhcp dhcpd.conf car tu y fais mention... 
> Est-ce que ça fait pas doublon avec reservation.conf ?
>
>

Ok pour /etc/sambaedu.

Il faut :
- sambaedu.conf (minimal). se4ad_name doit être spécifié, sinon il 
posera la question si il n'arrive pas à faire la résolution dns inverse
- sambaedu.conf.d/dhcp.conf ( le ficihier /etc/se3/config_d_cache.sh ) 
Surtout utile dans le cas de vlans pour récupérer les plages d'adresses.
- reservations.conf (dans un premier temps repris tel quel, puis ensuite 
le module dhcp ecrira les réservatinos dans le AD)
- les tables sql connexions et quotas
- les clés ssh root ( dans le cas où on migre depuis se3, il faut que la 
clé privée générée sur se3 puisse être copiée comme clé root de se4fs, 
et que la clé publique ait été mise en place sur se4ad )

Si tout ceci est fourni l'installation sera totalement silencieuse.

Côté se4ad, il faut créer le keytab www-sambaedu, ou alors je le fais en 
ssh dans web-common, c'est peut-être mieux car cela permet de 
reconfigurer sans devoir intervenir sur se4ad.

le se4ad doit être configuré comme serveur ntp, c'est plus logique. en 
revanche faut-il prévoir un mécanisme de test et de reconfiguration à 
distance depuis l'interface ?

**Comment by mollef** [comment]
La version actuelle du script fonctionne (paquet 0.23~6) et les paquets SambaEdu s'installent désormais correctement et de façon 100% silencieux.
Reste à régler 
* La conf du dhcp qui ne remonte pas à la bonne place donc le dhcp reste non configuré
* Déposer quotas.sql et connexion.sql dans /etc/sambaedu

**Comment by denis-b** [comment]
Oui, il faut mettre tout le dossier en 600. C'est fait plus tard dans les
paquets stretch, mais dans le cas d''une preiinstallation il faut le faire
avant.
